### ISSUE TITLE
Why my modelcheckpoint is not saving weights based on best val_accuracy?

### ISSUE THREAD

**Issue by mobassir94** [keras-team/keras#13966]
the code i tried in kaggle kernel for saving weights is this : 

```
model_path = '/kaggle/working/jigsawMultilingual.h5'
from tensorflow.keras.callbacks import EarlyStopping, ModelCheckpoint, LearningRateScheduler

checkpoint = ModelCheckpoint(model_path, monitor='val_accuracy', mode='max', save_best_only=True)
es = EarlyStopping(monitor='val_accuracy', mode='max', patience=2, 
                   restore_best_weights=True, verbose=1)
lr_callback = LearningRateScheduler(lrfn, verbose=1)

callback_list = [checkpoint, es, lr_callback]


%%time
N_STEPS = x_train.shape[0] // BATCH_SIZE
EPOCHS = 4
train_history = model.fit(
    train_dataset,
    steps_per_epoch=N_STEPS,
    validation_data=valid_dataset,
    callbacks=callback_list,
    epochs=EPOCHS
)

 # have a look at the training results : 

Train for 4414 steps, validate for 63 steps

Epoch 00001: LearningRateScheduler reducing learning rate to 1e-06.
Epoch 1/4
4414/4414 [==============================] - 2301s 521ms/step - loss: 0.3190 - auc: 0.9200 - val_loss: 0.2821 - val_auc: 0.9171

Epoch 00002: LearningRateScheduler reducing learning rate to 3.142857142857143e-06.
Epoch 2/4
4414/4414 [==============================] - 2090s 473ms/step - loss: 0.1909 - auc: 0.9718 - val_loss: 0.2581 - val_auc: 0.9249

Epoch 00003: LearningRateScheduler reducing learning rate to 5.285714285714285e-06.
Epoch 3/4
4414/4414 [==============================] - 2090s 473ms/step - loss: 0.1639 - auc: 0.9791 - val_loss: 0.2730 - val_auc: 0.9214

Epoch 00004: LearningRateScheduler reducing learning rate to 7.4285714285714275e-06.
Epoch 4/4
4414/4414 [==============================] - 2090s 473ms/step - loss: 0.1491 - auc: 0.9826 - val_loss: 0.2761 - val_auc: 0.9252
CPU times: user 9min 46s, sys: 35.5 s, total: 10min 21s
Wall time: 2h 22min 50s


```
but i  don't see any model in this path : model_path = '/kaggle/working/jigsawMultilingual.h5'

i mean jigsawMultilingual.h5 is not saving in '/kaggle/working/' directory

is it happening because validation accuracy is improving every time? but still it should save the  model  after 1st epoch,right?

**Comment by mzeeshanarshad** [comment]
I have the same issue. After the first epoch, there is no message at all regarding model saving. No error too but the file is not saved.

**Comment by innat** [comment]
@mobassir94 
Hi. I couldn't reproduce your issue either on kaggle (with `tf` 2.4.1) or clolab (with `tf` 2.6.0). The checkpoint works as expected in my case. If possible, please share a reproducible code. 

And from the above training log, I think your model never compiles with monitored metric; it seems like, only the `AUC` metric was used but the `ModelCheckpoint` is used to monitor the `Accuracy` metric. 

However, in your setup, the training logs would provide warnings like `WARNING:tensorflow: Can save the best model only with val_accuracy available, skipping.`

**Comment by jvishnuvardhan** [comment]
@innat Thanks.
@mobassir94 I couldn't reproduce the issue as mentioned by @innat . As the code was not complete, I took simple `mnist` model and was able to save weights based on best val_accuracy. Please check the [gist here](https://colab.research.google.com/gist/jvishnuvardhan/c05eb71c5cb10ccf9deafccfcbe30978/untitled.ipynb).

Try running with recent `TF2.6` or `tf-nightly`and share a simple standalone code if you are still facing this issue. Thanks!

**Comment by mobassir94** [comment]
hi @innat  @jvishnuvardhan 
sorry for this silly question,i am not very keen in tf,here is the code that i tried 1 year ago when i faced this issue : https://www.kaggle.com/mobassir/understanding-cross-lingual-models?scriptVersionId=31925922

in my recent tf models i don't face this issue anymore,thank you

**Comment by jvishnuvardhan** [comment]
Thanks for confirming. I will close the issue as this was resolved in recent TF. Thanks!
